appender_test.go

```diff
@@ -773,86 +776,25 @@ func TestAppenderFlushRequestError(t *testing.T) {
 		assert.Equal(t, int64(1), asserted.Load())
 	}
 	t.Run("400", func(t *testing.T) {
-		test(t, http.StatusBadRequest, "flush failed (400): {\"error\":{\"type\":\"x_content_parse_exception\",\"caused_by\":{\"type\":\"json_parse_exception\"}}}")
+		test(t, http.StatusBadRequest, "flush failed (400): "+string(error))
 	})
 	t.Run("403", func(t *testing.T) {
-		test(t, http.StatusForbidden, "flush failed (403): {\"error\":{\"type\":\"x_content_parse_exception\",\"caused_by\":{\"type\":\"json_parse_exception\"}}}")
+		test(t, http.StatusForbidden, "flush failed (403): "+string(error))
 	})
 	t.Run("429", func(t *testing.T) {
-		test(t, http.StatusTooManyRequests, "flush failed (429): {\"error\":{\"type\":\"x_content_parse_exception\",\"caused_by\":{\"type\":\"json_parse_exception\"}}}")
+		test(t, http.StatusTooManyRequests, "flush failed (429): "+string(error))
 	})
 	t.Run("500", func(t *testing.T) {
-		test(t, http.StatusInternalServerError, "flush failed (500): {\"error\":{\"type\":\"x_content_parse_exception\",\"caused_by\":{\"type\":\"json_parse_exception\"}}}")
+		test(t, http.StatusInternalServerError, "flush failed (500): "+string(error))
 	})
 	t.Run("503", func(t *testing.T) {
-		test(t, http.StatusServiceUnavailable, "flush failed (503): {\"error\":{\"type\":\"x_content_parse_exception\",\"caused_by\":{\"type\":\"json_parse_exception\"}}}")
+		test(t, http.StatusServiceUnavailable, "flush failed (503): "+string(error))
 	})
 	t.Run("504", func(t *testing.T) {
-		test(t, http.StatusGatewayTimeout, "flush failed (504): {\"error\":{\"type\":\"x_content_parse_exception\",\"caused_by\":{\"type\":\"json_parse_exception\"}}}")
+		test(t, http.StatusGatewayTimeout, "flush failed (504): "+string(error))
 	})
 }
 
-func TestAppenderIndexFailedLogging(t *testing.T) {
```

Comment:
this test cannot be removed now. We need to check that reason is sanitized on unset